### DISCUSSION TITLE
Help with initial set-up and folders

### DISCUSSION THREAD

**Discussion by TC72** [nvim-telekasten/telekasten.nvim#175] [Q&A]
Just dipping my toes into Zettelkasten and chose Telekasten as my preferred tool. As a programmer I've been living in Neovim/Vim/vi for almost 30 years and i've written a small extension for Telescope so it's a good fit for me.

I've created this folder structure:  
~/Documents/Reference  
  /daily  
  /weekly  
  /templates  
  /media

With this structure if I use `Telekasten find_notes` it includes all the .md files in daily, weekly and templates.
Is that the right way for it to work?

**Comment by lambtho12** [comment]
`find_notes` will look for all markdown files in the `home` that you specified in the config (including its sub-directories).
There is currently no option to the `find_note` function to tailor that behavior. But that would probably be something nice to look into.

Eventually it depends on your workflow... 

You could put stuff outside of your telekasten `ħome` directories, so it does not clutter your note taking (see below). With that you could still fairly easily open the daily/weekly notes using the ad-hoc functions. The problem would then be that `search_notes` (= grep in notes) will not look into these either. 
```
Notes
├──zettel
│   ├── 202104081704.md
│   ├── 202104081706.md
│   ├── 202104081733.md
│   ├── 202104081741.md
├── journal
│   ├── dailies
│       ├── 2022-10-10.md
│       ├── 2022-10-11.md
│   ├── weeklies
│       ├── 2022-10-10.md
├── templates
│   ├── daily.md
│   ├── newnotes.md
```
This works (for now) for me as I consider the journal and the notes as two completely separate stuff.

An other possibility would be to allow arguments to the lua functions, so when you setup your keybindings you could modify the behavior of the function. That would require a bit more work and I do not know how it would fit. The pickers all have a `search_dir` options (currently set to `M.Cfg.home`, the `home` directory specified in telekasten's config). We could make that one of the function's argument. Although, I do not know if it is possible to exclude some subdirs from the picker... If not, this might still require some tweaks in the way the directories are set-up (so zettels are clearly separated from journal and templates and media).

**Reply by lambtho12** [reply]
Unfortunately, I am very very busy at the moment. I doubt I will have time personally to make that happen before February.

**Reply by TC72** [reply]
Thanks for the reply and the thorough explanation. I'd love to get involved with helping this project. This might be a nice starting point.
I have a good grasp of Telescope, I wrote this extension: https://github.com/TC72/telescope-tele-tabby.nvim
I've also helped out with a couple of minor issues, including adding the ability to pass a function into `Telescope current_buffer_fuzzy_find` to decide the order matched lines are displayed.

**Reply by lambtho12** [reply]
Sure, please do! I will happily review your PR.

In the meantime, could you also please describe this feature in the issue tracker, so we can better keep track of it all? Github discussions tend to be very messy very quick in general 😅
